src/Diagnostics.HealthChecks/Readme.md

```diff
@@ -0,0 +1,105 @@
+
+The developer must keep in mind that an health check that includes the startup health check in its implementation will keep executing if the health check result results in a warning or in an error state; this can happen when the service the health check it's implemented into is created by Service Fabric in response to an automatic scale-up.
+
+The developer must also keep in mind that to use this composable Health Check they will have to provide an implementation of the [`IMemoryCache`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.imemorycache?view=dotnet-plat-ext-7.0) interface through Dependency Injection.
+
+#### [`HttpEndpointHealthCheck`](./Composables/HttpEndpointHealthCheck.cs)
+
+This implementation covers the most common usage of health checks. It will send a request to a controller endpoint in the context of the Service Fabric service (i.e. it will perform the call to `localhost`), it will handle the most common exceptions linked to HTTP calls using `HttpClient`, and it will eventually check the response for it to respond to the requirements.
+
+This health check offers extensibility through the definition of different functions:
+
+- `requestBuilder`: this function will have to return the request message that will be used by the `HttpClient` to perform the request.
+- `healthCheckResponseChecker`: this function will receive in input all the information about the HTTP call response; its responsibility will be to report the Health Check Status as a response, and it will be used to determine the result of the Health Check.
+
+In an effort to keep the Health Check as general as possible, the constructor will need an instance of the following:
+
+- `IHttpClientFactory`: it will be the responsibility of the implementor to provide a way to create the `HttpClient` instance. This will be addressed in the [Implementation examples](#implementation-examples).
+- `ActivitySource`: this Health Check will create an `Activity` instance. If a wrapper implementation in the composition classes chain created one already, it should be linked to the newly created one inside this health check.
+
+The `HttpEndpointHealthCheck` offers the possibility to "mark" the created `Activity` with custom markers by defining the `activityMarker` function in the constructor.
+
+### Implementation examples
+
+Implementing an Health Check using the composable classes provided by default can be done following two patterns:
+
+- **Creating the classes manually**: this approach consists on creating the classes manually inside the constructor of the Health Check that will be registered for the application. The resulting Health Check will have to declare the join set of dependencies needed by all the default classes used, along with its own.
```

Comment:
I'd suggest swapping the order of these - in the examples below too.
It seems like we'd generally want folks to use the extensions, unless they need the extra flexibility; so that approach should come first.